pontoon/base/utils.py

```diff
@@ -657,3 +657,91 @@ def build_translation_memory_file(creation_date, locale_code, entries):
         u'\n\t</body>'
         u'\n</tmx>'
     )
+
+
+def get_m2m_changes(current_qs, new_qs):
+    """
+    Get difference between states of m2m relation.
+    When formfield_items is empty, return all items from object_items as removed.
+
+    :arg django.db.models.QuerySet `current_qs`: objects from the current state of relation.
+    :arg django.db.models.QuerySet `final_qs`: objects from the future state of m2m
+    :returns: A tuple with 2 querysets for added and removed items from m2m
+    """
+
+    add_items = new_qs.exclude(
+        pk__in=current_qs.values_list('pk', flat=True)
+    )
+
+    remove_items = (
+        current_qs.exclude(
+            pk__in=new_qs.values_list('pk', flat=True)
+        ).exclude(
+            pk__in=add_items.values_list('pk', flat=True)
+        )
+    )
+
+    return list(add_items), list(remove_items)
+
+
+def log_group_members(manager, group, (add_users, remove_users)):
+    """Log all changes a group's members.
+
+    :arg django.contrib.auth.models.User manager: a manager who perform changes on groups.
+    :arg django.contrib.auth.models.Group group: group of translators/managers
+    :arg django.db.models.QuerySet add_groups: QuerySet with users to add
+    :arg django.db.models.QuerySet remove_groups: QuerySet with users to remove
+    """
+    from pontoon.base.models import PermissionChangelog
+
+    log_entries = [
+        PermissionChangelog(
+            action_type='added',
+            performed_by=manager,
+            performed_on=user,
+            group=group,
+        )
+        for user in add_users
+    ]
+
+    log_entries += [
+        PermissionChangelog(
+            action_type='removed',
+            performed_by=manager,
+            performed_on=user,
+            group=group,
+        )
+        for user in remove_users
+    ]
+    PermissionChangelog.objects.bulk_create(log_entries)
+
+
+def log_user_groups(admin, user, (add_groups, remove_groups)):
+    """Log all changes of a user's groups.
+
+    :arg django.contrib.auth.models.User admin: a super user who perform changes on groups.
+    :arg django.contrib.auth.models.User user: an edited users
+    :arg django.db.models.QuerySet add_groups: QuerySet with groups to add
+    :arg django.db.models.QuerySet remove_groups: QuerySet with groups to remove
+    """
+    from pontoon.base.models import PermissionChangelog
+
+    log_entries = [
+        PermissionChangelog(
+            action_type='added',
+            performed_by=admin,
+            performed_on=user,
+            group=group,
+        )
+        for group in add_groups
+    ]
+    log_entries += [
+        PermissionChangelog(
+            action_type='removed',
+            performed_by=admin,
+            performed_on=user,
+            group=group,
+        )
+        for group in remove_groups
+    ]
+    PermissionChangelog.objects.bulk_create(log_entries)
```

Comment:
The permissions page is part of `pontoon.teams` app. Shouldn't this code be there? I think we should keep `pontoon.base` only for stuff that's not app-specific.